### PR TITLE
[add package]: tilda

### PR DESCRIPTION
**Pull request by ELWAER-M**

### CODE REVIEW

File: packages/tilda/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+TERMUX_PKG_DESCRIPTION="A Gtk based drop down terminal for Linux and Unix"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@ELWAER-M"
+TERMUX_PKG_VERSION="1.5.0"
+TERMUX_PKG_SRCURL="https://github.com/lanoxx/tilda/archive/refs/tags/tilda-${TERMUX_PKG_VERSION}.tar.gz"
+TERMUX_PKG_SHA256="f664c17daca2a2900f49de9eb65746ced03c867b02144149ef21260cbcd61039"
+TERMUX_PKG_DEPENDS="glib, gtk3, libvte, libconfuse, libx11"
+
+termux_step_pre_configure() {
+	./autogen.sh --prefix=$TERMUX_PREFIX
```
Comment (xeffyr) [inline packages/tilda/build.sh:11]: `NOCONFIGURE=1 ./autogen.sh`

**Review by xeffyr — CHANGES_REQUESTED** [review]

**Comment by Yisus7u7** [comment]
@ELWAER-M You should not close this until it merges.

File: packages/tilda/build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+TERMUX_PKG_HOMEPAGE="https://github.com/lanoxx/tilda"
+TERMUX_PKG_DESCRIPTION="A Gtk based drop down terminal for Linux and Unix"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@ELWAER-M"
+TERMUX_PKG_VERSION="1.5.0"
+TERMUX_PKG_SRCURL="https://github.com/lanoxx/tilda/archive/refs/tags/tilda-${TERMUX_PKG_VERSION}.tar.gz"
+TERMUX_PKG_SHA256="f664c17daca2a2900f49de9eb65746ced03c867b02144149ef21260cbcd61039"
+TERMUX_PKG_DEPENDS="glib, gtk3, libvte, libconfuse, libx11, gettext, automake, autoconf"
```
Comment (xeffyr) [inline packages/tilda/build.sh:8]: Remove gettext, automake and autoconf. They are not needed as runtime dependencies and not used during build time. Packages are cross-compiled and build tools installed in Termux prefix are not used and can't be anyway.

File: packages/tilda/build.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+TERMUX_PKG_EXTRA_MAKE_ARGS="--silent"
```
Comment (xeffyr) [inline packages/tilda/build.sh:9]: Why passing `--silent` to make? There no reason to make build logs shorter. They are intended to be informative.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/tilda/build.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+
+termux_step_pre_configure() {
+    NOCONFIFURE=1 ./autogen.sh
```
Comment (xeffyr) [inline packages/tilda/build.sh:13]: Typo there. Should be `NOCONFIGURE=1 ./autogen.sh`, i.e. not `NOCONFIFURE=1 ...`.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/tilda/build.sh (hunk 2 above)
Comment (xeffyr) [reply packages/tilda/build.sh:8]: Well, gettext maybe still necessary.

**Comment by xeffyr** [comment]
There few more issues than mentioned in review:
1. Need `TERMUX_PKG_BUILD_IN_SRC=true`
2. Compilation error:
```
src/tilda.c:57:10: fatal error: 'sys/dir.h' file not found
#include <sys/dir.h>
```

**Review by xeffyr — CHANGES_REQUESTED** [review]
As I have already mentioned, there is a compilation error (missing header) which needs a fix:
```
  CC       src/tilda-key_grabber.o
  CC       src/tilda-screen-size.o
  CC       src/tilda-tilda.o
  CC       src/tilda-tilda-keybinding.o
src/tilda.c:57:10: fatal error: 'sys/dir.h' file not found
#include <sys/dir.h>
         ^~~~~~~~~~~
1 error generated.
src/tilda-keybinding.c:306:43: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
                                          _ ("Enter keyboard shortcut"));
                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/data/com.termux/files/usr/include/glib-2.0/glib/gi18n.h:26:20: note: expanded from macro '_'
#define  _(String) gettext (String)
```

**Comment by ELWAER-M** [comment]
just forget that

**Pull request closed without merging**